src/merge.ts

```diff
@@ -0,0 +1,43 @@
+import { EventPayload } from './EventPayload';
+
+/**
+ * Used for merging request JSON into an existing EventPayload.
+ * Returns a new EventPayload, with the original object unaffected.
+ * The merge occurs using the following conventions:
+ *
+ *  Merging an existing key with null/undefined deletes the key.
+ *  Merging an existing key with non-null value updates the value.
+ *  Merging a non-existing key adds the key/value.
+ *  When the value is an object, the above is applied iteratively.
+ */
+export const merge = (original: EventPayload, newValues: PartialPayload): EventPayload => {
+  const copy = { ...original };
+  const stack: Array<[PartialPayload, PartialPayload]> = [[copy, newValues]];
+
+  while (stack.length) {
+    const [target, source] = stack.pop() ?? [];
+    if (target && source) {
+      Object.keys(source).forEach((key) => {
+        const EventKey = key as keyof EventPayload;
```

Comment:
seems a bit odd to capitalize EventKey when it's a variable and not a type or constructor